testutils/source.go

```diff
@@ -679,7 +678,40 @@ func main() {
 	log.Printf("Waiting for command to finish...")
 	err = cmd.Wait()
 	log.Printf("Command finished with error: %v", err)
-}`}, 1, gosec.NewConfig()}}
+}`}, 1, gosec.NewConfig()}, {[]string{`
+// syscall.Exec function called with harcoded arguments
+// shouldn't be consider as a command injection
+package main
+import (
+	"fmt"
+	"syscall"
+)
+func main() {
+	err := syscall.Exec("/bin/cat", []string{"/etc/passwd"}, nil)
+	if err != nil {
+		fmt.Printf("Error: %v\n", err)
+	}
+}`}, 0, gosec.NewConfig()},
+		{[]string{`
+// starting a process with a variable as an argument
```

Comment:
Please could you add one more test for the case when a variable is a function argument. Something like:

```Go
func RunCmd(args ...strings) {
     cmd := exec.Command("cmd", args...)
     err := cmd.Start()
     if err != nil {
          log.Fatal(err)
     }
}

fun main() {
      RunCmd(os.Args...)
}
```

```

Reply:
While I was working on this new test I noticed this issue:
https://github.com/securego/gosec/issues/343

Reply:
Yeah, this needs more advanced data flow analysis in order to reliably solve it. It should recourse down the graph to find the origin of the initialisation.

I think for now, it's fine if there is a warning in this case and we let the user to investigate whether or not is a vulnerability.

Reply:
So, do I need other unit tests or something?

Reply:
It would be nice to handle this case in the rule and return a warning. Also it would be great to have a unit test for this which checks that in this case the rule returns a warning.

Reply:
@MVrachev Are you still working on this? I would love to get this fix merged. I hope we can do a release soon, and it would be nice to include also this one. Thanks!

Reply:
Yes, I haven't abandoned it. It's just I have to finish other work.
I will do my best to finish it this week. 
Sorry for the inconvenience.

Reply:
Excellent!

Reply:
For this code:

```
package main

import (
	"log"
	"os/exec"
)

func RunCmd(command string) {
	cmd := exec.Command(command, "5")
	err := cmd.Start()
	if err != nil {
		log.Fatal(err)
	}
	log.Printf("Waiting for command to finish...")
	err = cmd.Wait()
}

func main() {
	RunCmd("sleep")
}
```
Gosec will return the following warning: `G204: Subprocess launched with variable (Confidence: HIGH, Severity: MEDIUM)
  > exec.Command(command, "5")`

What you want me to do is just to add a unit test for this instance?

Reply:
Yeah, it would be great to add this code sample to the list of samples tested on G204 here https://github.com/securego/gosec/blob/98749b7357954795b616fc335c98fb7bcb144576/testutils/source.go#L616. Thanks!

Reply:
Done. I updated my pr with the new unit test.